src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -2111,7 +2111,26 @@ private static bool CheckIfMethodIsApplicableAndPrepareIt(MethodData method, Exp
 			promotedArgs.AddRange(method.Parameters.Skip(promotedArgs.Count).Select<ParameterInfo, Expression>(x =>
 			{
 				if (x.HasDefaultValue)
-					return Expression.Constant(x.DefaultValue, x.ParameterType);
+				{
+					var parameterType = x.ParameterType;
+					if (x.ParameterType.ContainsGenericParameters)
+					{
+						//try finding an actual parameter for the generic
+						for(var i = 0; i < promotedArgs.Count; i++)
+						{
+							var actualArg = promotedArgs[i];
```

Comment:
Nitpick: this could be moved to line 2124 since it's not used in the `if` condition.

Reply:
True